controllers/status.go

```diff
@@ -240,6 +242,58 @@ func (r *PerformanceProfileReconciler) getKubeletConditionsByProfile(profile *pe
 	return r.getDegradedConditions(conditionKubeletFailed, latestCondition.Message), nil
 }
 
+func (r *PerformanceProfileReconciler) getTunedConditionsByProfile(profile *performancev2.PerformanceProfile) ([]conditionsv1.Condition, error) {
+	tunedProfileList := &tunedv1.ProfileList{}
+	if err := r.List(context.TODO(), tunedProfileList); err != nil {
+		klog.Errorf("Cannot list Tuned Profiles to match with profile %q : %v", profile.Name, err)
+		return nil, err
+	}
+
+	selector := labels.SelectorFromSet(profile.Spec.NodeSelector)
+	nodes := &corev1.NodeList{}
+	if err := r.List(context.TODO(), nodes, &client.ListOptions{LabelSelector: selector}); err != nil {
+		return nil, err
+	}
+
+	// remove Tuned profiles that are not associate with this perfomance profile
+	// Tuned profile's name and node's name should be equal
+	filtered := removeUnMatchedTunedProfiles(nodes.Items, tunedProfileList.Items)
+	message := bytes.Buffer{}
+	for _, tunedProfile := range filtered {
+		isDegraded := false
+		isApplied := true
+		var tunedDegradedCondition *tunedv1.ProfileStatusCondition
+
+		for _, condition := range tunedProfile.Status.Conditions {
+			if (condition.Type == tunedv1.TunedDegraded) && condition.Status == corev1.ConditionTrue {
+				isDegraded = true
+				tunedDegradedCondition = &condition
+			}
+
+			if (condition.Type == tunedv1.TunedProfileApplied) && condition.Status == corev1.ConditionFalse {
+				isApplied = false
+			}
+		}
+		// We need both condition to exists,
+		// since there is a scenario where both Degraded & Applied condition are true
```

Comment:
I am curious when can you have degraded true and applied true?

Reply:
You should refer this question to @yanirq .
But as far as I understood there is a situation when tuned configuration applied with errors. then is will be both applied and degraded as true

Reply:
This is something that should be eventually fixed under tuned. tuned has a buffer that keeps errors and this buffer is read by NTO.  
When NTO detects errors in that buffer it updates the profile state accordingly but it could get to a state where a profile was successfully applied  but the tuned error buffer was not cleaned up , therefore NTO  sets both status as true (degraded + applied)  that is why we should have our status degraded only when degraded = true and applied = false